compiler/src/iree/compiler/Codegen/LLVMGPU/ConvertToROCDL.cpp

```diff
@@ -123,8 +125,11 @@ struct ConvertToROCDLPass final
       RewritePatternSet patterns(&getContext());
       // These patterns only convert a subset of arith that target specific
       // rocdl intrinsics (e.g. fp8 conversions).
+      bool convertFP8Arithmetic = (*maybeChipset).majorVersion == 9 &&
+                                  (*maybeChipset).minorVersion >= 0x40;
       arith::populateArithToAMDGPUConversionPatterns(
-          patterns, /*saturateFP8Truncf=*/false);
+          patterns, convertFP8Arithmetic, /*saturateFP8Truncf=*/false,
+          /*allowPacked16Rtz=*/false, maybeChipset.value_or(amdgpu::Chipset()));
```

Comment:
Why use `value_or` if you dereference it before?

Reply:
Good point. I just used what was there upstream blindly. Fixed.